pkg/asset/tls/ironictls.go

```diff
@@ -0,0 +1,47 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/sirupsen/logrus"
+)
+
+// IronicTLSCert is the asset that generates the key/cert pair that is used for
+// enabling TLS for virtual media in ironic.
+type IronicTLSCert struct {
+	SignedCertKey
+}
+
+var _ asset.WritableAsset = (*IronicTLSCert)(nil)
+
+// Dependencies returns the dependency of the the cert/key pair, which includes
+// the parent CA, and install config if it depends on the install config for
+// DNS names, etc.
+func (a *IronicTLSCert) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&RootCA{},
+	}
+}
+
+// Generate generates the cert/key pair based on its dependencies.
+func (a *IronicTLSCert) Generate(ctx context.Context, dependencies asset.Parents) error {
+	ca := &RootCA{}
```

Comment:
Looking at the [equivalent implementation in CBO](https://github.com/openshift/cluster-baremetal-operator/blob/release-4.16/provisioning/baremetal_crypto.go#L56-L88), there (as in previous versions of this patch) we generate a self-signed CA that signs the ironic cert (and is then itself discarded - we use a new CA every time we refresh the certificate).
I think signing with the root CA could be a mistake, because it will encourage users to configure their BMCs to trust the root CA (and by extension, all of the certificates that may be directly or indirectly signed by it in the future, which is a large surface area of unrelated things).

Of course at the moment users will have to disable host verification, which is worse, because we don't actually provide the CA cert anywhere in CBO. Presumably what we should really do (but is out of scope of this bug) is to store the self-signed CA cert + key in the cluster and reuse the same one in CBO, plus make the CA cert available to the user so they can configure BMCs to trust it (or does ironic do this already?). I think having an independent self-signed CA will put us on a path to eventually implementing that.

Reply:
Okay, so we need to change SignedCertKey to SelfSignedCertKey on line 15 and drop ca from the Generate call, right?

Reply:
I have now created a self-signed CA, just like in the CBO implementation.

Reply:
Done

Reply:
> Okay, so we need to change SignedCertKey to SelfSignedCertKey on line 15 and drop ca from the Generate call, right?

No, in CBO we generate a self-signed CA and then a signed cert from that CA.